kedro-org/kedro-viz
# Refactor / node list row

## Description



## Development notes
Initially, the node-list-row component was used in three different contexts:

- `node-list-tree`: (labelled as 3 in the chart below) representing the list of nodes
- `node-list-group`: (labelled as 4A) is the parent level within the filter panel (e.g., Element types, Tags)
- `node-row-list`: (labelled as 4B) is the child level within the filter panel (e.g., nodes, datasets, parameters)
<img width="1108" alt="Screenshot 2024-10-17 at 11 53 30" src="https://github.com/user-attachments/assets/8ca6d52a-b03c-4787-8a16-8792896b7c8a">


The new approach involves splitting node-list-tree into two distinct components:

- `row`: dedicated to the hierarchical structure of node elements
- `filter-row`: dedicated to the filter panel
- Additionally, a `ToggleControl` component is shared between these two components. This simple component renders an icon with the toggle functionality to turn on/off the functionality

![Screenshot 2024-10-24 at 10 54 38](https://github.com/user-attachments/assets/1f2575f2-b1da-4842-ad74-b390ed2ab011)

## QA notes
Please check the changes locally and test throughout to see if any changes that would affect the functionality. I'm currently updating some of the cypress tests which currently failed due to changes in classnames. 

## Checklist
